src/modules/wreck/lua.d/spectrum.lua

```diff
@@ -0,0 +1,47 @@
+-- Set environment specific to spectrum_mpi (derived from openmpi)
+--
+
+if wreck:getopt ("mpi") ~= "spectrum" then return end
+
+local posix = require 'posix'
+
+function prepend_path (env_var, path)
+    local env = wreck.environ
+    if env[env_var] == nil then
+       suffix = ''
+    else
+       suffix = ':'..env[env_var]
+    end
+    env[env_var] = path..suffix
+end
+
+function rexecd_init ()
+    local env = wreck.environ
+    local f = wreck.flux
+    local rundir = f:getattr ('broker.rundir')
+
+    -- Avoid shared memory segment name collisions
+    -- when flux instance runs >1 broker per node.
+    env['OMPI_MCA_orte_tmpdir_base'] = rundir
+
+    env['OMPI_MCA_osc'] = "pt2pt"
+    env['OMPI_MCA_pml'] = "yalla"
+    env['OMPI_MCA_btl'] = "self"
+    env['MPI_ROOT'] = "/opt/ibm/spectrum_mpi"
+    env['OPAL_LIBDIR'] = "/opt/ibm/spectrum_mpi/lib"
+    env['OMPI_MCA_coll_hcoll_enable'] = '0'
+
+    env['PMIX_SERVER_URI'] = nil
+    env['PMIX_SERVER_URI2'] = nil
+
+    -- help find libcollectives.so
+    prepend_path ('LD_LIBRARY_PATH', '/opt/ibm/spectrum_mpi/lib/pami_port')
+    prepend_path ('LD_PRELOAD', '/opt/ibm/spectrum_mpi/lib/libpami_cudahook.so')
+end
+
+function rexecd_task_init ()
+    -- approximately ulimit -Ss 10240
+    posix.setrlimit ("stack", 10485760)
```

Comment:
If we know a version of lua.posix that this doesn't work on, perhaps note it here in a comment?

Also perhaps note that this is set to shut up the spectrum MXM plugin.

Reply:
I will look into seeing exactly when luaposix switched this interface to deprecated and add a conditional that uses the newer interface on newer luaposix libraries.

And I will add the comment about the spectrum plugin

Reply:
If you look at `src/bindings/lua/flux/posix.lua` we already have a wrapper around the luaposix module to "work around" at least one compat issue.

You could use the same `pcall()` trick  (used in there to check for `posix.time`) to check for `posix.sys.resource` and use that module's setrlimit() instead of the one from deprecated.lua.

Reply:
Though now that I think about it I wonder if `flux.posix` is loaded in wrexecd context... have to check on that

Reply:
I was going down the route of parsing the luaposix version string and specializing based on that version, when I realized `posix.setrlimit ("stack", 10485760, -1)` always does the right thing.  As long as we are OK using this deprecated function, there is no need for any of the version string parsing nastiness, etc.  Thoughts?

Reply:
Does that fail if the current hard limit isn't unlimited?

Reply:
Spoke with @grondo and @garlick in person.  For the sake of getting this out the door, the existing code should be fine.  We can revisit if any problems in the wild pop up.